File: examples/benchmarks/primes2.pl
#! perl

my $i=0;
my $max=5000;
my $i6 = 0;
my $i7;

while (1) {
    if (&isprime1($i)) {
	$i7 = $i;
	$i6++;
    }
    $i++;
    if ($i==$max){
	last;
    }
}
printf("N primes calculated to %d is %d\nlast is: %d\n",$max,$i6,$i7);

sub isprime1
{
    my($input)=@_;
    my $n;

    if ($input < 1) {
	return 0;
    }
    $n = $input - 1;

    while ($n > 1){
	if ($input % $n == 0) { return 0; }
	$n--;
    }
    return 1;
}

File: examples/benchmarks/primes2.rb
#! ruby

i6 = 0
i7 = 0
max = 5000

def is_prime(n)
	if n < 1
		return false
	end

	(n-1).downto(2) { |i|
		if n % i == 0
			return false
		end
	}
	true
end

max.times { |i|
	if is_prime i
		i6 += 1
		i7 = i
	end
}
puts "# of primes calculated to #{max}: #{i6}"
puts "last is: #{i7}"

